### ISSUE TITLE
aws_iam_server_certificate crash

### ISSUE THREAD

**Issue by trevorr** [hashicorp/terraform#11010]
I'm getting the following crash/panic in 0.8.2 using `aws_iam_server_certificate`:

```
data.aws_iam_server_certificate.cert: Refreshing state...
Error refreshing state: 1 error(s) occurred:

* data.aws_iam_server_certificate.cert: unexpected EOF
panic: interface conversion: interface is string, not bool
```

Here is the entire TF source:

```
provider "aws" {
  profile = "prod"
  region = "us-east-1"
}

data "aws_iam_server_certificate" "cert" {
  name_prefix = "MyCert"
  latest = true
}
```

See the [crash.log](https://github.com/hashicorp/terraform/files/683080/crash.log.txt) for more details.

Note the crash only occurs when the `name_prefix` actually matches a cert. The expected, non-panic error occurs when no certs match (`Search for AWS IAM server certificate returned no results`).

**Comment by trevorr** [comment]
Looks like [line of 50 of `builtin/providers/aws/data_source_aws_iam_server_certificate.go`](https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/data_source_aws_iam_server_certificate.go#L50) should be `TypeBool` instead of `TypeString`:

```
			"latest": &schema.Schema{
				Type:     schema.TypeString,
				Optional: true,
				ForceNew: true,
				Default:  false,
			},
```

**Comment by stack72** [comment]
Hi @trevorr 

I will submit a PR to get this fixed for 0.8.3 - sorry for the hassle here! Thanks for helping us identify the bug

Paul

**Comment by trevorr** [comment]
Thanks. BTW, I made a dev build with that change, and it works correctly for me. I'd send a PR, but you could probably write the test case much faster/better than I could. 😉
